Composer/packages/client/src/components/Adapters/TeamsManifestGenerator.tsx

```diff
@@ -0,0 +1,131 @@
+import { Text } from 'office-ui-fabric-react/lib/Text';
+import { IButtonStyles, IconButton } from 'office-ui-fabric-react/lib/components/Button';
+import { FontSizes, NeutralColors } from '@uifabric/fluent-theme/lib/fluent';
+import { useRef } from 'react';
+import { ITextField, TextField } from 'office-ui-fabric-react/lib/components/TextField';
+import { DialogTypes, DialogWrapper } from '@bfc/ui-shared/lib/components/DialogWrapper';
+import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
+
+import { defaultTeamsManifest } from '../../constants';
+
+const iconButtonStyle: IButtonStyles = {
+  root: {
+    height: 'unset',
+    float: 'right',
+    marginRight: '10px',
+  },
+  menuIcon: {
+    backgroundColor: NeutralColors.white,
+    color: NeutralColors.gray130,
+    fontSize: FontSizes.size16,
+  },
+  rootDisabled: {
+    backgroundColor: NeutralColors.white,
+  },
+  rootHovered: {
+    backgroundColor: 'unset',
+  },
+  rootPressed: {
+    backgroundColor: 'unset',
+  },
+};
+
+type TeamsManifestGeneratorProps = {
+  hidden: boolean;
+  botAppId?: string;
+  botDisplayName?: string;
+  onDismiss: () => void;
+};
+
+export const TeamsManifestGenerator = (props: TeamsManifestGeneratorProps) => {
+  const textFieldRef = useRef<ITextField>(null);
+
+  const copyCodeToClipboard = () => {
+    try {
+      textFieldRef.current?.select;
+      textFieldRef.current?.select();
+      document.execCommand('copy');
+      textFieldRef.current?.setSelectionRange(0, 0);
+      textFieldRef.current?.blur();
+    } catch (e) {
+      console.error('Something went wrong when trying to copy manifest content to clipboard.', e);
+    }
+  };
+
+  const generateTeamsManifest = () => {
+    const appId = props.botAppId ? props.botAppId : '{AddBotAppId}';
+    const botName = props.botDisplayName ? props.botDisplayName : '{AddBotDisplayName}';
+    const result = defaultTeamsManifest;
+    result.id = uuidv4().toString();
+    result.description.short = `short description for ${botName}`;
+    result.description.full = `full description for ${botName}`;
+    result.packageName = botName;
+    result.name.short = botName;
+    result.name.full = botName;
+    result.bots[0].botId = appId;
+    return JSON.stringify(defaultTeamsManifest, null, 2);
+  };
+
+  return (
+    <DialogWrapper
+      dialogType={DialogTypes.CreateFlow}
+      isOpen={!props.hidden}
+      subText={formatMessage(
+        'Your Teams adapter is configured for your published bot. Copy the manifest, open App Studio in Teams and add the manifest so you can test your bot in Teams'
+      )}
+      title={formatMessage('Teams Manifest')}
+      onDismiss={props.onDismiss}
+    >
+      <div>
+        <Text style={{ fontWeight: 700 }}>{formatMessage('Teams manifest for your bot:')}</Text>
+        <a
+          download={'teamsManifest.json'}
+          href={'data:text/plain;charset=utf-8,' + encodeURIComponent(generateTeamsManifest())}
+        >
+          <IconButton
```

Comment:
why is this an icon button if it's wrapped by an anchor tag?

Reply:
removed